introduce initial version of CI description for oneDAL (#223)
uxlfoundation/open-source-working-group: project-infrastructure/project-ci-documentation.md

oneDAL
------

Representative: Nikolay Petrov[https://github.com/napetrov]

Support contacts for CI:

| Area               | Owner                                                                                           |
| ------------------ | ----------------------------------------------------------------------------------------------- |
| AArch64            | Hamza Butt[https://github.com/theComputeKid] Rakshith G B[https://github.com/rakshithgb-fujitsu]|
| RISC-V             | Keeran Rothenfusser[https://github.com/keeranroth]                                              |
| x64, new platforms | Nikolay Petrov[https://github.com/napetrov]                                                     |

More details available in - https://github.com/uxlfoundation/oneDAL/blob/main/MAINTAINERS.md

*Existing public CI*

oneDAL

| Platform | Type | OS | Number | Active? | Comments |
| --- | --- | --- | --- | --- | --- |
| Github | CPU AArch64 | Linux | 2 | Yes | 2 Pipelines on Physical ARM systems Github hosted and ARM hosted. ci-aarch64.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci-aarch64.yml]  |

> | Github | Intel GPU (ICX compiler) | Linux | 1 | Yes* | Pipelene configured for GPU valdiation on uxlfoundation GPU runners, currently disabled due to issues with Tiber Cloud. ci.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci.yml]  |

Reviewer: typo: Pipelene, valdiation

| Github | ABI conformance | Linux | 1 | Yes | 2 Pipelines on Physical ARM systems Github hosted and ARM hosted. ci.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci.yml]  |
| Github | Docker validation | Linux | 1 | Yes | Validation of oneDAL development env docker file. docker-validation-ci.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/docker-validation-ci.yml]  |

> | Github | CPU Nightly | Linux, Windwos | 2 | Yes | Nightly builds and broaer validation for oneDAL. nightly-build.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/nightly-build.yml]  |

Reviewer: There was some talk on separating the Nightly CI from those needed for PR acceptance: https://github.com/uxlfoundation/open-source-working-group/pull/218#discussion_r2037348901

Author: Interesting - this puts whole another question on separation between PR level checks, CI level and scheduled tasks

| Azure DevOps | CI build/test for x86/ARM/RISC-V with OS compilers | Linux, Windows | 8 | Yes | Intel build natively, ARM and RISC-V with crosplatform build and QEMU emulation. Build with GCC/VC compilers. ci.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.ci/pipeline/ci.yml]  |
| Azure DevOps | CI build/test with Bazel | Linux | 1 | Yes | Bazel based build and validation. ci.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.ci/pipeline/ci.yml]  |
| Azure DevOps | oneDAL documentation build | Linux | 1 | Yes | Build documentation for validation purposese in CI. docs.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.ci/pipeline/docs.yml]  |
| Azure DevOps | sklearnex validation | Linux | 1 | Yes | Checking out sklearnex sources, doing build and validation with oneDAL PR changes. ci.yml[https://github.com/uxlfoundation/oneDAL/blob/main/.ci/pipeline/ci.yml]  |
| Codefactor | Codefactor checks | N/A | 1 | Yes | Enforcing code checks in PRs, Bandit, and other code quality checks. Codefactor[https://www.codefactor.io/repository/github/uxlfoundation/onedal]  |

*Required Public CI Infrastruture Needed To Confidently Accept Contributions*

> Currently internal intel CI validation is requred for code verification. 

Reviewer 2: Any use in summarizing the scope of internal CI? You say below public covers ~20%, but should we explain the gaps?

```suggestion
Currently internal intel CI validation is required for code verification. 
```

Intel employees can start internal CI with comment "/intelci: run" and corresponding CI would be triggered.
It's not possible to access logs without access to Intel network

> Migration to public validation is possible but would requere more x86 systems availabe in public - currenlty public validation covers only ~20% in validation scopes 

Reviewer: typo: requere, currenlty

oneDAL
------

Representative: [Nikolay Petrov](https://github.com/napetrov)

Support contacts for CI:

| Area               | Owner                                                                                               |
| ------------------ | --------------------------------------------------------------------------------------------------- |
| AArch64            | [Hamza Butt](https://github.com/theComputeKid) [Rakshith G B](https://github.com/rakshithgb-fujitsu)|
| RISC-V             | [Keeran Rothenfusser](https://github.com/keeranroth)                                                |
| x64, new platforms | [Nikolay Petrov](https://github.com/napetrov)                                                       |

More details available in [MAINTAINERS.md](https://github.com/uxlfoundation/oneDAL/blob/main/MAINTAINERS.md)

### *Existing public CI*

oneDAL

| Platform | Type | OS | Number | Active? | Comments |
| --- | --- | --- | --- | --- | --- |

> | Github | CPU AArch64 | Linux | 2 | Yes | 2 Pipelines on Physical ARM systems Github hosted and ARM hosted. [ci-aarch64.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci-aarch64.yml)  |

Author: ```suggestion
| Github | CPU AArch64 | Linux | 2 | Yes | 2 Pipelines on Physical Arm systems Github hosted and Arm-Hosted. [ci-aarch64.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci-aarch64.yml)  |
```

| Github | Intel GPU (ICX compiler) | Linux | 1 | Yes* | Pipeline configured for GPU validation on uxlfoundation GPU runners, currently disabled due to issues with Tiber Cloud. [ci.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci.yml)  |

> | Github | ABI conformance | Linux | 1 | Yes | 2 Pipelines on Physical ARM systems Github-hosted and ARM-hosted. [ci.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci.yml)  |

Reviewer: "Arm" systems, "Arm-Hosted"

Same in the above line.

Author: thanks - and actually this was incorrect copy-paste for this line. ABI checks done on X86 systems currently.  But i've fixed actual Arm line

> | Github | ABI conformance | Linux | 1 | Yes | 2 Pipelines on Physical ARM systems Github-hosted and ARM-hosted. [ci.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci.yml)  |

Author: ```suggestion
| Github | ABI conformance | Linux | 1 | Yes | 2 Pipelines on Physical Arm systems Github-hosted and Arm-Hosted. [ci.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/ci.yml)  |
```

| Github | Docker validation | Linux | 1 | Yes | Validation of oneDAL development env docker file. [docker-validation-ci.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/docker-validation-ci.yml)  |
| Github | CPU Nightly | Linux, Windows | 2 | Yes | Nightly builds and broader validation for oneDAL. [nightly-build.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/nightly-build.yml)  |
| Github | Copyright headers check | Linux | 1 | Yes | Check for proper copyright headers. [skywalking-eyes.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/skywalking-eyes.yml)  |
| Github | PR checklist validation | Linux | 1 | Yes | Validation of PR conformance. [pr-checklist.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/pr-checklist.yml)  |
| Github | Documentation deployment to gh-pages | Linux | 1 | Yes | Automatic docs deployment with release tag creation. [docs-release.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/workflows/docs-release.yml)  |

> | Mergify | Helper automation for merges/backporting | Linux | 1 | Yes | Automated labels assignment, removal of renovate branches. .[mergify.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/.mergify.yml)  |

Reviewer 3: ```suggestion
| Mergify | Helper automation for merges/backporting | Linux | 1 | Yes | Automated labels assignment, removal of renovate branches. [.mergify.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.github/.mergify.yml)  |
```

| Azure DevOps | CI build/test with Bazel | Linux | 1 | Yes | Bazel-based build and validation. [ci.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.ci/pipeline/ci.yml)  |
| Azure DevOps | oneDAL documentation build | Linux | 1 | Yes | Build documentation for validation purposes in CI. [docs.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.ci/pipeline/docs.yml)  |
| Azure DevOps | sklearnex validation | Linux | 1 | Yes | Checking out sklearnex sources, doing build and validation with oneDAL PR changes. [ci.yml](https://github.com/uxlfoundation/oneDAL/blob/main/.ci/pipeline/ci.yml)  |
| Codefactor | Codefactor checks | N/A | 1 | Yes | Enforcing code checks in PRs, Bandit, and other code quality checks. [Codefactor](https://www.codefactor.io/repository/github/uxlfoundation/onedal)  |

scikit-learn-intelex

| Platform | Type | OS | Number | Active? | Comments |
| --- | --- | --- | --- | --- | --- |

> | Mergify | Helper automation for merges/backporting | Linux | 1 | Yes | Automated labels assignment, removal of renovate branches. .[mergify.yml](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/.github/.mergify.yml)  |

Reviewer 3: ```suggestion
| Mergify | Helper automation for merges/backporting | Linux | 1 | Yes | Automated labels assignment, removal of renovate branches. [.mergify.yml](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/.github/.mergify.yml)  |
```

| Github | PR checklist validation | Linux | 1 | Yes | Validation of PR conformance. [pr-checklist.yml](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/.github/workflows/pr-checklist.yml)  |
| Github | Documentation deployment to gh-pages | Linux | 1 | Yes | Automatic docs deployment with release tag creation. [docs-release.yml](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/.github/workflows/docs-release.yml)  |
| Codefactor | Codefactor checks | N/A | 1 | Yes | Enforcing code checks in PRs, Bandit, and other code quality checks. [Codefactor](https://www.codefactor.io/repository/github/uxlfoundation/scikit-learn-intelex)  |
| Codecov | Codecoverage | N/A | 1 | Yes | Enforcing coverage tracking and increments in PRs. [Codecov](https://app.codecov.io/gh/uxlfoundation/scikit-learn-intelex)  |

### *Required Public CI Infrastructure Needed To Confidently Accept Contributions*

Currently internal Intel CI validation is required for code verification. 
Intel employees can start internal CI with comment "/intelci: run" and corresponding CI would be triggered.

> It's not possible to access logs without access to Intel network

Reviewer 3: ```suggestion
It's not possible to view these logs without access to Intel network
```
just for avoiding double use of "access"

> Migration to public validation is possible but would require more x86 systems available in public - currently public validation covers only ~20% of validation scopes 

Reviewer 3: @ahuber21 made a point that this might be a little intimidating, maybe something like:
```suggestion
Migration to public validation is possible but will require more x86 systems available in public - currently public validation covers a small subset of validation scopes that we are planning to expand
```
Or something along those lines
